include/query.php

```diff
@@ -185,4 +193,31 @@ public function filter_query( $lang ) {
 			}
 		}
 	}
+
+	/**
+	 * Sets the language correctly if the current query is a 'OR' relation,
+	 * since WordPress merges the language with the other query vars when the relation is OR.
+	 *
+	 * @since 3.3
+	 *
+	 * @return void
+	 */
+	protected function set_language_for_or_relation() {
+		if ( $this->query->tax_query instanceof WP_Tax_Query
+			&& 'OR' === $this->query->tax_query->relation
+			&& isset( $this->query->tax_query->queried_terms['language'] )
+			&& 'all' !== $this->query->tax_query->queried_terms['language']['terms'] ) {
```

Comment:
Is this condition useful? It looks to me that next lines will remove 'all' as anyway.